tests/nightly/test_large_array.py

```diff
@@ -274,11 +265,40 @@ def test_diag():
     assert_almost_equal(r.asnumpy(), np.diag(a_np, k=k))
 
     # random k
-    k = np.random.randint(-min(LARGE_X, 64) + 1, min(h, w))
+    k = np.random.randint(-min(LARGE_X, SMALL_Y) + 1, min(LARGE_X, SMALL_Y))
     r = mx.nd.diag(a, k=k)
     assert_almost_equal(r.asnumpy(), np.diag(a_np, k=k))
 
 
+def random_2d_coordinates(x_low, x_high, y_low, y_high):
```

Comment:
I just realized there is existing utility that does what you want to achieve here. Please use https://github.com/apache/incubator-mxnet/blob/master/python/mxnet/test_utils.py#L418